mobile/screens/Training/ViewSingleLogScreen.tsx

```diff
@@ -58,6 +64,44 @@ export default function ViewSingleLogScreen(props: any) {
     loadLogInformation().then().catch();
   }, []);
 
+  const getFullDownloadPath = (path: string) => {
+    return FileSystem.documentDirectory + path;
+  };
+
+  const ensureDownloadDirectoryExists = async (path: string) => {
+    if (!path.includes("/")) {
+      return;
+    }
+
+    const directory = path.substring(0, path.lastIndexOf("/"));
+
+    await FileSystem.makeDirectoryAsync(getFullDownloadPath(directory), {
+      intermediates: true,
+    });
+  };
+
+  const downloadVideo = async () => {
+    setDownloadState(DownloadState.InProgress);
+
+    const download = FileSystem.createDownloadResumable(
+      videoUrl,
+      getFullDownloadPath(video),
+      {}
+    );
+
+    await ensureDownloadDirectoryExists(video);
+    const file = await download.downloadAsync();
+
+    if (!file) {
+      setDownloadState(DownloadState.NotStarted);
+      throw new Error("Failed to download video");
+    }
+
+    await MediaLibrary.saveToLibraryAsync(file.uri);
```

Comment:
This throws an error if the user does not grant access to add photos. Catch that and throw an error letting the user know that it failed to save the video to photos and they need to grant access to add photos in settings.